ai/movie-recommendation/src/main/java/com/hazelcast/samples/ai/movies/TextSimilaritySearchImap.java

```diff
@@ -0,0 +1,364 @@
+/*
+ * Copyright (c) 2008-2024, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hazelcast.samples.ai.movies;
+
+import com.hazelcast.client.HazelcastClient;
+import com.hazelcast.config.Config;
+import com.hazelcast.config.InMemoryFormat;
+import com.hazelcast.config.MapConfig;
+import com.hazelcast.core.Hazelcast;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.core.Offloadable;
+import com.hazelcast.internal.util.Timer;
+import com.hazelcast.map.EntryProcessor;
+import com.hazelcast.map.IMap;
+import com.hazelcast.query.Predicate;
+import com.hazelcast.query.Predicates;
+import dev.langchain4j.model.embedding.onnx.allminilml6v2.AllMiniLmL6V2EmbeddingModel;
+import dev.langchain4j.model.embedding.EmbeddingModel;
+
+import javax.annotation.Nullable;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.ObjectInputStream;
+import java.io.Reader;
+import java.io.Serial;
+import java.io.Serializable;
+import java.io.StringReader;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.Map;
+import java.util.Scanner;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.regex.Pattern;
+
+/**
+ * This demo generates embeddings of movie plot summaries from the
+ * <a href="https://www.cs.cmu.edu/~ark/personas/">"CMU Movies Summary Corpus"</a>. The embeddings
+ * are stored in an IMap, along with movie metadata (title, release date).
+ * We then perform similarity search on user's input strings from console and list top 10 movies
+ * with most similar plot summaries.
+ *
+ * <p>Dependencies:</p>
+ * <ul>
+ *     <li>langchain4j for the <a href="https://huggingface.co/sentence-transformers/all-MiniLM-L6-v2">all-miniLM-L6-v2</a>
+ *     model that transforms text to 384 dimensional dense vector</li>
+ * </ul>
+ */
+public class TextSimilaritySearchImap {
+
+    static boolean offloaded = true;
```

Comment:
can you write a quick comment what it means "offloaded" in this particular case? Will be easier to understand for people.

Reply:
https://github.com/hazelcast/hazelcast-code-samples/pull/710/commits/a82b3b5e44e968a2793b991e982a9dd3111df44d